plugins/utils/trimmer/copy.c

```diff
@@ -91,67 +94,43 @@ enum bt_component_status update_packet_context_field(FILE *err,
 	struct bt_ctf_field *field = NULL, *writer_field = NULL;
 	int nr_fields, i, int_ret;
 
+	BT_LOGI("Updating packet context field: name=%s", name);
 	packet_context = bt_ctf_packet_get_context(writer_packet);
-	if (!packet_context) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__, __FILE__,
-				__LINE__);
-		goto error;
-	}
+	assert(packet_context);
 
 	struct_type = bt_ctf_field_get_type(packet_context);
-	if (!struct_type) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__, __FILE__,
-				__LINE__);
-		goto error;
-	}
+	assert(struct_type);
 
 	writer_packet_context = bt_ctf_packet_get_context(writer_packet);
-	if (!writer_packet_context) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__, __FILE__,
-				__LINE__);
-		goto error;
-	}
+	assert(writer_packet_context);
 
 	nr_fields = bt_ctf_field_type_structure_get_field_count(struct_type);
 	for (i = 0; i < nr_fields; i++) {
 		const char *field_name;
 
 		field = bt_ctf_field_structure_get_field_by_index(
 				packet_context, i);
-		if (!field) {
-			fprintf(err, "[error] %s in %s:%d\n", __func__,
-					__FILE__, __LINE__);
-			goto error;
-		}
+		assert(field);
 		if (bt_ctf_field_type_structure_get_field(struct_type,
 					&field_name, &field_type, i) < 0) {
-			fprintf(err, "[error] %s in %s:%d\n", __func__,
-					__FILE__, __LINE__);
+			BT_LOGE("Failed to get field: field_name=%s", field_name);
 			goto error;
 		}
 		if (strcmp(field_name, name)) {
 			BT_PUT(field_type);
 			BT_PUT(field);
 			continue;
 		}
-		if (bt_ctf_field_type_get_type_id(field_type) != BT_CTF_FIELD_TYPE_ID_INTEGER) {
-			fprintf(err, "[error] Unexpected packet context field type\n");
-			goto error;
-		}
+		assert(bt_ctf_field_type_get_type_id(field_type) ==
```

Comment:
I am expecting the timestamp_begin/end fields here, if they are not integers, the plugin won't be able to work, but indeed it is more an input problem, I will report that as an error instead of an assert